pymess/backend/sms/ats_sms_operator.py

```diff
@@ -230,30 +230,32 @@ def _update_sms_states_from_response(self, messages, parsed_response, is_sending
                 )
 
     def publish_messages(self, messages):
+        messages = (message for message in messages if self._is_recipient_phone_number_allowed(message))
         self._send_requests(messages, request_type=self.REQUEST_TYPES.SMS, is_sending=True, sent_at=timezone.now())
```

Comment:
I don't understand why this method is overriden anyway. In parent, it just calls `publish_message` for each message. Here, it is doing the same thing as `publish_message` does, so what's the point.

Reply:
this `publish_messages` method sends all messages in one request. Otherwise it would do one request per message.